iroh/src/endpoint.rs

```diff
@@ -682,21 +694,10 @@ impl Endpoint {
                 .endpoint()
                 .connect_with(client_config, addr.socket_addr(), "localhost")?;
 
-        let connection = connect
-            .await
-            .context("failed connecting to remote endpoint")?;
-
-        let rtt_msg = RttMessage::NewConnection {
-            connection: connection.weak_handle(),
-            conn_type_changes: self.conn_type(node_id)?.stream(),
-            node_id,
-        };
-        if let Err(err) = self.rtt_actor.msg_tx.send(rtt_msg).await {
```

Comment:
why is this not needed anymore?

Reply:
Because by wrapping `quinn::Connecting` with `iroh::endpoint::Connecting`, we run `try_send_rtt_msg` in `impl Future for Connecting`.